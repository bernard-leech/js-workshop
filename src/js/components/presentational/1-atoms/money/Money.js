// @flow
import React from 'react'

// import type { MoneyType } from './Money.types'

function numberWithSeparator(number, separator) {
    return number.toString().replace(/\B(?=(\d{3})+(?!\d))/g, separator);
}

function Money (props) {
  const {
    moneyValue,
    currencySymbol = 'Â£',
    separator = ',',
    decimal = '.'
  } = props
  
  const integerValue = Math.floor(moneyValue)
  const fractionalValue = Math.round((moneyValue - integerValue) * 100)
  const integerValueWithCommas = numberWithSeparator(integerValue, separator);
  
  const money = `${currencySymbol}${integerValueWithCommas}${decimal}${fractionalValue}`
  
  return (
    <div className='number' style={{fontVariantNumeric: 'tabular-nums', fontFamily: '-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif'}}>
      {money}
    </div>
  )
}

export default Money
