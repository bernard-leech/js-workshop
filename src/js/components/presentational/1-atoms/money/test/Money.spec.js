import React from 'react'
import TestRenderer from 'react-test-renderer'

import { props } from '../test/Money.mock'

import Money from '../Money'

it('renders correctly', () => {
  const tree = TestRenderer
    .create(<Money {...props} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
