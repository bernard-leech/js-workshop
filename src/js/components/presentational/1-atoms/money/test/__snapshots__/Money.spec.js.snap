// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="number"
  style={
    Object {
      "fontFamily": "-apple-system,BlinkMacSystemFont,\\"Segoe UI\\",Roboto,Oxygen-Sans,Ubuntu,Cantarell,\\"Helvetica Neue\\",sans-serif",
      "fontVariantNumeric": "tabular-nums",
    }
  }
>
  Â£1,234.56
</div>
`;
